### PR TITLE
Add option to improve frame pacing through duplicate frames if below 60hz.

### LINKED ISSUE #12325: Frameskipping cause glitches with shader that have requires60fps
### What happens?
If have shader have 60FPS enabled when one unthrottle the game speed the game show old frame, stop completely on one or loop over some.
It's graphic only, the game keep running normally and one unthrottle is stopped the graphic get back to normal as well.

Sometimes the shader get totally stuck and display nothing (0 fps, all black), when this happen stop unthrottle doesn't fix it and the shader must be turned off to let the game render again.

Only shader that have this is "CTR scanline" (note that "Scanline (CTR)" require no 60fps and work fine).

### What *should* happen?
Either should be fixed, ignore the 60 fps requirement and let the shader bug its timing (better that the totally corrupted graphic), or just disable the shader during unthrottle.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.
Vulkan and OpenGL are affected on both Android and desktop linux. I only had MHFU to test that have a max framerate of 30, might be related.

### PR DESCRIPTION
**Pull request by hrydgard**
Should help #9736, and partially papers over #12325, and should help #12460 .

### CODE REVIEW

File: UI/GameSettingsScreen.cpp [hunk 1]
```diff
@@ -345,6 +345,11 @@ void GameSettingsScreen::CreateViews() {
 #ifdef _WIN32
 	graphicsSettings->Add(new CheckBox(&g_Config.bVSync, gr->T("VSync")));
 #endif
+	CheckBox *frameDuplication = graphicsSettings->Add(new CheckBox(&g_Config.bRenderDuplicateFrames, gr->T("Render duplicate frames to 60hz")));
+	frameDuplication->OnClick.Add([=](EventParams &e) {
+		settingInfo_->Show(gr->T("RenderDuplicateFrames Tip", "Can make framerate smoother in games that run at 30hz"), e.v);
```
Comment (LunaMoo) [inline UI/GameSettingsScreen.cpp]: Maybe ".. in games that run under 60hz"? Since it will also affect 20 and other weird fps PSP games have.

**Review by LunaMoo — COMMENTED** [review]
Seeing iota97 https://github.com/hrydgard/ppsspp/issues/12325#issuecomment-579676197 it seems disabling the flag on unthrottle might not be enough:c.

 I wonder, maybe the actual problem causing this was also causing the bug with recording from "Current Framebuffer" in games running under 60fps as it looked similar ~ recorded old frames every other one.

**Comment by hrydgard** [comment]
Yeah, there still seems to be some kind of gremlin in the machinery related to this...

**Comment by mirh** [comment]
I know that 60hz is the standard refresh rate, but perhaps tooltips should be a bit more "general" than that.

**Comment by hrydgard** [comment]
Yeah, fair enough. Changed it again.

**Comment by LunaMoo** [comment]
It's whatever, but to make it clear. 60hz here came from about PSP being limited to max 60hz and the option affecting only things < 60hz. This option is not going do display differently on modern displays than it would on 60hz ones.

 If PPSSPP will ever allow more ie to support for example 120 fps patches, it might then differ, but currently it's not and most games would break horribly when patched to such framerates even if working fine with 60fps patch(and that would already be limited, since most of those patches break stuff).

**Comment by hrydgard** [comment]
I think mirh was referring more to people maybe not knowing that 60hz is the standard PSP refresh rate. Though then again, if you don't know that, you might not be interested in the option anyway. Not sure which wording makes more sense.

**Comment by iota97** [comment]
I think it's better to say something like: "game with frame rate capped below 60"

Saying running lower might end up with user enabling this when the HW can't keep up with 60 fps. Not sure if it may help there as well tho'.

**Comment by mirh** [comment]
I was actually thinking to the computer display... 
If for whatever reason I'm natively running at 30hz (which could also happen during normal operation with some variable refresh ratio monitor?), duplicating frames doesn't make sense regardless.

**Review by unknownbrackets — APPROVED** [review]
This may also have issues with alt speed over 100%, since that's partially unthrottled.  Maybe when alt speed is above 100% we auto-disable?

Also, created #12665 for the flickering issue.

-[Unknown]

**Comment by hrydgard** [comment]
Rebased on master, resolving the merge conflict.

**Comment by hrydgard** [comment]
Rebased, merging shortly. I don't see much of an interesting improvement myself, but it might trigger various devices and drivers into better behavior if lucky I guess. So let's merge.

**Comment by unknownbrackets** [comment]
Also, this fixes #12460 right?

-[Unknown]

**Comment by hrydgard** [comment]
Yeah, changed the initial post so it'll get linked.

File: Core/HLE/sceDisplay.cpp [hunk 2]
```diff
@@ -754,9 +754,11 @@ void __DisplayFlip(int cyclesLate) {
 	// Also let's always flip for animated shaders.
 	const ShaderInfo *shaderInfo = g_Config.sPostShaderName == "Off" ? nullptr : GetPostShaderInfo(g_Config.sPostShaderName);
 	bool postEffectRequiresFlip = false;
+	// postEffectRequiresFlip is not compatible with frameskip unthrottling, see #12325.
 	if (shaderInfo && g_Config.iRenderingMode != FB_NON_BUFFERED_MODE)
-		postEffectRequiresFlip = shaderInfo->requires60fps;
+		postEffectRequiresFlip = (shaderInfo->requires60fps || g_Config.bRenderDuplicateFrames) && !(g_Config.bFrameSkipUnthrottle && !FrameTimingThrottled());
```
Comment (unknownbrackets) [inline Core/HLE/sceDisplay.cpp]: Oops, the setting only applies if `shaderInfo` is not nullptr.

-[Unknown]

**Pull request merged**